cvxpy/lin_ops/canon_backend.py

```diff
@@ -873,8 +873,6 @@ def parametrized_mul(x):
         else:
             assert len(lhs) == 1
             reps = view.rows // lhs[0].shape[-1]
-            if isinstance(lhs[0], sp.coo_matrix):
```

Comment:
I removed these instance checks which were temporary fixes for the shape incompatibilities in mul and al. 
Since we added _do_dense helper method below, this if statement is useless.